shell-script/public-gateway-checker.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/bash
+
+# IPFS public gateway checker v1.0 (shell script version)
+# original repository: https://github.com/ipfs/public-gateway-checker
+# this script added by Joss Brown: https://github.com/JayBrown
+
+JSON_URL="https://raw.githubusercontent.com/ipfs/public-gateway-checker/master/gateways.json"
+TESTHASH="Qmaisz6NMhDB51cCvNWa1GMS7LU1pAxdF4Ld6Ft9kZEP2a"
+
+echo "Public IPFS Gateways"
+
+GATEWAYS=$(curl --silent "$JSON_URL" | sed -e '$ d' -e '1,1d' -e 's/\"//g')
+if [[ $GATEWAYS == "" ]] ; then
+	echo "Error: no gateways found."
+	exit
+fi
+
+GWNUM=$(echo "$GATEWAYS" | wc -l | xargs)
+
+while read -r GATEWAY
+do
+
+	GATEWAY=$(echo "$GATEWAY" | xargs)
+	TESTURL=$(echo "$GATEWAY" | sed -e "s-:hash,-$TESTHASH-" -e "s-:hash-$TESTHASH-")
+	DOMAIN=$(echo "$GATEWAY" | awk -F/ '{print $1"//"$3}')
+
+	if [[ $(curl --silent "$TESTURL") == "Hello from IPFS Gateway Checker" ]] ; then
```

Comment:
what about using the option `-L` to follow redirects?

Reply:
Hm, would better to fix the URLs to be correct from the beginning instead, but be more future-proof to follow redirect so I agree.

Reply:
Use case: you run the script and `awk -F"Online:" '{print $2}'` to get the domains, then you won't have the redirect domain, but only the domain of the original request. If you want to do something with the gateway domain, e.g. automate a file upload if it's writable or simply cache an IPFS object via cURL, then running the script without an `-L` option would just tell you that something isn't right with the gateway URL. This would in turn benefit the canonical list.

But in this case, too, we could add a redirect option `-r`. Then the script would just need to print the final URL, not the initial one.